### DISCUSSION TITLE
Spreading water in Landlab erosion model applications

### DISCUSSION THREAD

**Discussion by gregtucker** [landlab/landlab#1970] [General]
In May 2024 I received the following email, which prompted this discussion thread:

"I am Clément Garcia-Estève, Doctor in Geosciences at University of Montpellier in France.
My thesis defense was in October 2023 and my subject was How extreme tectonic and climatic events affect landscape morphology ?
With my thesis supervisors Rodolphe Cattin (University of Montpellier) and Stéphane Dominguez (CNRS), we have developed a new approach combining analog and numerical modeling, especially using Landlab.

We are currently improving the numerical model to have a morphology similar to the analog model. Analog experiments and numerical simulations involve applying discrete tectonic events and climatic cycles to a landscape over a period of around 50,000 years. However, we have a limitation due to the flow accumulator in Landlab. I put a synthesis of my question in attachment.
In the numerical model at the outlet of each watershed, only one drain has the entire water flow. As a result, the valley width is reduced and incision is greatly amplified in this area compared to the analog model.
To fix this limitation, we'd like to spread the flow of water over several cells surrounding a drain during flood periods. This would have the effect of reducing incision and creating a wider valley similar to the analog model and to nature.
We hope you can answer our question. Yours faithfully.

Garcia-Estève Clément"

**Comment by gregtucker** [comment]
The problem as I understand it lies in a limitation of the oft-used "single direction" water-routing method for erosion models. This is the algorithm applied, for example, when you use the Landlab FlowAccumulator with the FlowDirectorSteepest or FlowDirectorD8 option. Water is routed from each cell to just one downstream neighbor. Therefore the flow can never be more than one grid cell wide.

As background, it's important to remember that a cell-to-cell routing scheme, like the one that FlowAccumulator implements, represents one of many possible methods for calculating water flow across a terrain surface. Routing algorithms like this are basically a computationally efficient way to solve the steady form of the shallow-water equations. The shallow-water equations are, in turn, a vertically integrated form of the more general Navier-Stokes equations with the terrain surface as a boundary condition. So the broader question is: what's the best way to calculate flow over a terrain surface for your particular application, i.e., numerically modeling the physical erosion experiment? Clearly, single-direction routing is too simple, so what is the "best" way to capture the spread of water while keeping reasonable computational efficiency? (Note: Greg Hancock and I wrote a bit about different flow methods in Tucker & Hancock, 2010 ESPL).

I can think of at least three options:

1) Use a 2D dynamic flow model that numerically solves the shallow-water equations, or an approximation thereof. One relevant Landlab component is called OverlandFlow. It solves a form of the dynamic shallow-water equations in which the convective acceleration terms are dropped. The numerical method was developed by Bates et al. (2010) and refined by de Almeida et al. (2012). @mcflugen has recently done a lot of work to refactor its code and improve speed. An advantage of this approach: it gets closer to the relevant fluid physics, and because it includes gravity, pressure, and inertial terms, flow will spread out. Potential disadvantages: it's dynamic rather than steady, and requires small time-steps (however, if the time scale of physical experiments is hours rather than the millennia we usually deal with in landscape evolution, that might not be a problem). It also does not work with many of Landlab's fluvial landscape evolution components, as these require a specified flow direction. However, there are at least two simple components (DepthSlopeProductEroder and DetachmentLtdErosion) that are designed to work with flow solvers like OverlandFlow (and Yuval Shmilovitz is working on a much more sophisticated one for cliff and gully erosion; see Shmilovitz et al. (2024 JGR)).

2) Use a flow-routing algorithm that has multiple flow directions. The FlowAccumulator provides options for multiple flow routing, including FlowDirectorDinf (the D-infinity algorithm) and FlowDirectorMFD (flow divided among downstream cells in proportion to gradient). The drawback here is that the Landlab fluvial landscape evolution components are not YET configured to handle multiple downstream flow directions. They can and should be! But someone needs to do the requisite engineering (I started but the code is in a heap on the shop floor, so to speak...). The simplest approach is to record, for each cell, both an array of downstream slopes and discharges, and a single value that represents the steepest path among them. One could use the steepest of those paths (and its corresponding discharge) as the basis for computing erosion/transport rate, while retaining the ability to have the flow spread out across multiple cells downstream. The concept is simple but it does require some re-engineering, and that re-engineering would need to maintain backward compatibility.

3) If your erosion law is simple, you could just write a new erosion component that is specifically designed to handle multiple flow directions.

Have others run into similar issues with flow routing? If so, how have you handled them?

**Reply by Hailesh10** [reply]
Hi Prof. Tucker,

I am reaching out to seek your guidance regarding the use of Landlab for modeling riverbank erosion. Specifically, I am interested in incorporating the following components into my model: Groundwater interactions, freezing and thawing (Ice model), Fluvial erosion and Mass wasting. Could you kindly provide any advice, resources, or suggestions on how to best approach integrating these elements into a one model using Landlab? Additionally, if there are any specific modules or documentation you would recommend, I would greatly appreciate your recommendations. Thank you for your time and assistance. I look forward to your insights and advice on this project. Many thanks in advance!

Warm regards,

**Reply by gregtucker** [reply]
It's hard to provide advice without knowing more about your research questions, but in general I think it's important to first formulate a clear concept of your system, then derive equations to describe it using mass, momentum, and/or energy balance. Maybe you have done all this already? Once you have your equation set, then you can think about how to calculate solutions using numerical methods implemented in Landlab. A great guide to these steps is R. Slingerland and L. Kump (2011) Mathematical modeling of earth's dynamical systems.

**Reply by Hailesh10** [reply]
Thank you for the clarifications and sending the imortanet document through! I have formulated my research questions ( to see the effect of those process on streambank erosion). I wanted to integrate three to four existing componet of the landlab to test on the permafrost streambank erosion in unstructured 2D mesh grid considering the aformesniond processes. which landlab components are useful in this regard?Thank you for the support!

> It's hard to provide advice without knowing more about your research questions, but in general I think it's important to first formulate a clear concept of your system, then derive equations to describe it using mass, momentum, and/or energy balance. Maybe you have done all this already? Once you have your equation set, then you can think about how to calculate solutions using numerical methods implemented in Landlab. A great guide to these steps is R. Slingerland and L. Kump (2011) Mathematical modeling of earth's dynamical systems.

Thank you for the clarifications and sending the imortanet document through! I have formulated my research questions ( to see the effect of those process on streambank erosion). I wanted to integrate three to four existing componet of the landlab to test on the permafrost streambank erosion in unstructured 2D mesh grid considering the aformesniond processes. which landlab components are useful in this regard?Thank you for the support!

**Reply by gregtucker** [reply]
Josie Arcuri at CU Boulder has worked on modeling permafrost streambank erosion using code written with Landlab. You might wish to contact her.
